### PR TITLE
User form is created by factory now.

### PR DESCRIPTION
**Pull request by ikovalyov**

### CODE REVIEW

File: src/Controller/Factory/UserControllerFactory.php [hunk 1]
```diff
@@ -18,6 +18,6 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
         $userManager = $container->get(UserManager::class);
         
         // Instantiate the controller and inject dependencies
-        return new UserController($entityManager, $userManager);
+        return new UserController($entityManager, $userManager, $container);
     }
 }
```
Comment (Xapp73) [inline src/Controller/Factory/UserControllerFactory.php]: empty line plz if possible

**Comment by ikovalyov** [comment]
[Add user management system. Full information + API keys](https://trello.com/c/e9tG1LLH/15-add-user-management-system-full-information-api-keys)

**Comment by ikovalyov** [comment]
[Add user management system. Full information + API keys](https://trello.com/c/e9tG1LLH/15-add-user-management-system-full-information-api-keys)

File: src/Controller/Factory/UserControllerFactory.php (hunk 1 above)
Comment (HesyRa) [inline src/Controller/Factory/UserControllerFactory.php]: new line

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -386,6 +391,30 @@ public function getUserRole($user)
...
+    {
+        $result = $this->userManager->generateToken();
+        return new JsonModel(['result' => $result]);
+    }
+
+    /**
+     * Used to override form fields
+     *
+     * @param $form
+     * @param $user
```
Comment (Xapp73) [inline src/Controller/UserController.php]: Correct annots plz (missing type)

**Review by Xapp73 — CHANGES_REQUESTED** [review]

**Review by Xapp73 — APPROVED** [review]

**Review by HesyRa — APPROVED** [review]

**Review by HesyRa — APPROVED** [review]

**Review by Xapp73 — APPROVED** [review]

File: src/Controller/UserController.php [hunk 3]
```diff
@@ -211,6 +213,16 @@ public function editAction()
     }
 
     /**
+     * @param $id
+     * @return null|object
```
Comment (Xapp73) [inline src/Controller/UserController.php:217]: User?

File: src/Controller/UserController.php [hunk 4]
```diff
@@ -211,6 +213,16 @@ public function editAction()
     }
 
     /**
+     * @param $id
```
Comment (Xapp73) [inline src/Controller/UserController.php:216]: type?

**Review by Xapp73 — APPROVED** [review]

**Review by HesyRa — APPROVED** [review]

**Pull request merged**